File: frontend/src/app/features/epargne/models/compte-epargne-response.ts
import { StatutCompte } from "./compte-epargne-create-request";
import { TypeCompteEpargne } from "./type-compte-epargne";

export interface CompteEpargneResponse {
  id: number;
  membreId: number;
  membreNomComplet: string;
  numeroCompte: string;

  typeCompte: TypeCompteEpargne; // ✅ CORRIGÉ
  soldeDisponible: number;
  soldeBloque: number;

  statut: StatutCompte; // ✅ CORRIGÉ

  dateOuverture: string;
  dateFermeture?: string;

  createdAt: string;  // ✅ pas optionnel
  updatedAt: string;  // ✅ pas optionnel
}

File: backend/src/main/java/com/mini/credit/dto/epargne/CompteEpargneResponse.java
package com.mini.credit.dto.epargne;

import com.mini.credit.enums.StatutCompte;
import com.mini.credit.enums.TypeCompteEpargne;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Builder
public class CompteEpargneResponse {
    private Long id;
    private Long membreId;
    private String membreNomComplet;
    private String numeroCompte;
    private TypeCompteEpargne typeCompte;
    private BigDecimal soldeDisponible;
    private BigDecimal soldeBloque;
    private StatutCompte statut;
    private LocalDate dateOuverture;
    private LocalDate dateFermeture;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}